Feature/otp screens

baraa0abd opened the pull request:
## Description
implement otp screens with some fixes in login screen for code reusability 


## What Changed
- make a background widget for reusing it ( login , create acc , forget password and OTP)
- refactor the login based on the 1st point
- implement both OTP screens in forget password and create acc screen

## Screenshots/Videos (if applicable)
## CREATE ACCOUNT OTP ##


https://github.com/user-attachments/assets/0b34d78d-507c-431d-89a2-328db843b709



## FORGET PASSWORD OTP ##


https://github.com/user-attachments/assets/4e7c9552-73e9-4ee7-9a62-4f192261e795




## Checklist
- [x] My code follows the project's style guidelines
- [x] Changes have been tested manually and verified.
- [x] PR includes at most one single feature.

baraa0abd committed:
feat: Implement forget password OTP screen

baraa0abd committed:
refactor: Improve Forget Password OTP screen UI

baraa0abd committed:
Merge remote-tracking branch 'origin/develop' into feature/OTP-screens

baraa0abd committed:
Merge remote-tracking branch 'origin/develop' into feature/OTP-screens

baraa0abd committed:
Merge branch 'develop' into feature/OTP-screens

baraa0abd committed:
refactor: Extract AuthBackgroundWrapper from Login screen

Extracted the common background and layout structure from the Login screen into a reusable `AuthBackgroundWrapper` widget. This wrapper handles the background image, logo, and the overlapping container for authentication forms.

Additionally, this change introduces the new `signupOTP.dart` screen for account confirmation, which utilizes the new wrapper.

baraa0abd committed:
fix: Adjust spacing and layout on signup OTP screen

baraa0abd committed:
refactor: Remove signup confirmation logic and loading state

ghost commented:
<!-- sellio-bot-autoupdate-comment -->
**[Sellio Bot] Auto-update Status**


✅ Branch successfully updated from `develop`.

baraa0abd committed:
refactor: Style ForgetPasswordOTP screen

ghost commented:
<!-- sellio-bot-autoupdate-comment -->
**[Sellio Bot] Auto-update Status**


✅ Branch successfully updated from `develop`.

baraa0abd committed:
refactor: Remove comments from login screen

ghost commented:
<!-- sellio-bot-autoupdate-comment -->
**[Sellio Bot] Auto-update Status**


✅ Branch successfully updated from `develop`.

mohamedebrahem13 reviewed (changes requested):
![Screenshot_20251028_084442](https://github.com/user-attachments/assets/a00ece05-2a45-498c-b98e-d9a3590c8e7b)
There is extra padding from top in the white container 
The top section part is bigger should be smaller like the design

mohamedebrahem13 commented:
Also the otp card when select should be like design 
![Screenshot_20251028_084842](https://github.com/user-attachments/assets/80b3877d-9bca-4a18-9e2f-3aa812121098)

baraa0abd commented:
> ![Screenshot_20251028_084442](https://private-user-images.githubusercontent.com/111383089/506378375-a00ece05-2a45-498c-b98e-d9a3590c8e7b.jpg?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NjE2MzUwMTMsIm5iZiI6MTc2MTYzNDcxMywicGF0aCI6Ii8xMTEzODMwODkvNTA2Mzc4Mzc1LWEwMGVjZTA1LTJhNDUtNDk4Yy1iOThlLWQ5YTM1OTBjOGU3Yi5qcGc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjUxMDI4JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI1MTAyOFQwNjU4MzNaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT0wNDZiZjE1NjYzZTAzYmFjZTY2YTlkMjdiNTJkMTdhNWNkYmQ2MWFiOGEwMWYyN2ZiMDM2NGJkMjI1NGMwZjVjJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.tZ4qX254yw5pjTEEFsl6H8scoHee6B27AZV3zQtauHU) There is extra padding from top in the white container The top section part is bigger should be smaller like the design

##########################

I think it's the same as shown in the given img:
<img width="554" height="558" alt="image" src="https://github.com/user-attachments/assets/aab3699d-345f-4e9a-872f-479099b4ed8f" />

baraa0abd commented:
> Also the otp card when select should be like design ![Screenshot_20251028_084842](https://private-user-images.githubusercontent.com/111383089/506379900-80b3877d-9bca-4a18-9e2f-3aa812121098.jpg?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NjE2MzUwMTMsIm5iZiI6MTc2MTYzNDcxMywicGF0aCI6Ii8xMTEzODMwODkvNTA2Mzc5OTAwLTgwYjM4NzdkLTliY2EtNGExOC05ZTJmLTNhYTgxMjEyMTA5OC5qcGc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjUxMDI4JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI1MTAyOFQwNjU4MzNaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT0xMzc2NzY0YjMwZDBjYzJmZjAwYjhmOTlmMDE4MDlmNmZhN2IzZTNlZmRiMGMyYTMxOTA2NjFlMDcwZWU0MTg2JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.P83hQcIBZB7VXRqZCbWyv7RxzwcgdRNMjdE27vHLbxU)

Actually , it's same as figma ( as it was in the previous PR ) but the emulator quality screen make it looks like this

baraa0abd committed:
fix: Adjust padding and spacing on OTP screen

ghost commented:
<!-- sellio-bot-autoupdate-comment -->
**[Sellio Bot] Auto-update Status**


✅ Branch successfully updated from `develop`.

baraa0abd reviewed (commented):
## HERE IS THE UPDATED CHANGES ##

<img width="255" height="566" alt="image" src="https://github.com/user-attachments/assets/a828fda2-0cfa-47c0-a396-22cf031ed70a" />

baraa0abd committed:
feat: Add AuthBackground color and apply it

A new `AuthBackground` color has been added to the `SellioColorScheme`.
This new color is now used as the background color for the `AuthBackgroundWrapper` widget, replacing the previous `primary` color.

ghost commented:
<!-- sellio-bot-autoupdate-comment -->
**[Sellio Bot] Auto-update Status**


✅ Branch successfully updated from `develop`.

mohamedebrahem13 reviewed (dismissed):
great work!

YaseenOmar commented on lib/ui/screens/auth/forgetpasswordOTP.dart:
```diff
@@ -0,0 +1,259 @@
+                  ),
+                  child: Text(
+                    'Confirm',
+                    style: TextStyle(
```
use text theme from design system

YaseenOmar commented on lib/ui/screens/auth/signupOTP.dart:
```diff
@@ -0,0 +1,211 @@
+              alignment: Alignment.centerLeft,
+              child: Text(
+                'Confirm your account',
+                style: TextStyle(
```
use text theme

YaseenOmar commented on lib/ui/screens/auth/signupOTP.dart:
```diff
@@ -0,0 +1,211 @@
+              ),
+              child: Text(
+                'Confirm',
+                style: TextStyle(
```
use text theme

baraa0abd committed:
refactor: Use design system typography in OTP screens

ghost commented:
<!-- sellio-bot-autoupdate-comment -->
**[Sellio Bot] Auto-update Status**


✅ Branch successfully updated from `develop`.

YaseenOmar reviewed (approved):
Well done , Baraa

ghost commented:
<!-- sellio-bot-autoupdate-comment -->
**[Sellio Bot] Auto-update Status**


✅ Branch successfully updated from `develop`.
